### PR TITLE
Bump prometheus operator and prometheus

### PR DESCRIPTION
**Pull request by s-urbaniak**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
This bumps prometheus-operator to its newest version v0.46.0 and is a simplified version of #1661

### CODE REVIEW

**Comment by wojtek-t** [comment]
@s-urbaniak - the failure are related to this change (prometheus stack not starting correctly).
Can you please take a look?

File: clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml [hunk 1]
```diff
@@ -4,28 +4,34 @@ apiVersion: apps/v1
...
       - args:
-        - --logtostderr=true
-        - --config-reloader-image=gcr.io/k8s-testimages/quay.io/coreos/configmap-reload:v0.0.1
-        - --prometheus-config-reloader=gcr.io/k8s-testimages/quay.io/coreos/prometheus-config-reloader:v0.30.0
         {{if $PROMETHEUS_SCRAPE_KUBELETS}}
         - --kubelet-service=kube-system/kubelet
         {{end}}
-        image: gcr.io/k8s-testimages/quay.io/coreos/prometheus-operator:v0.31.0
+        - --prometheus-config-reloader=gcr.io/k8s-testimages/quay.io/prometheus-operator/prometheus-config-reloader:v0.46.0
+        image: gcr.io/k8s-testimages/quay.io/prometheus-operator/prometheus-operator:v0.46.0
```
Comment (s-urbaniak) [inline clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:34]: @wojtek-t it seems those images are not available yet. Do you have pointers on how these can be bumped within gcr.io?

Comment (mborsz) [reply clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:34]: I think it's manual process. I have copied the following images to gcr.io:
* gcr.io/k8s-testimages/quay.io/prometheus/prometheus:v2.25.0
* gcr.io/k8s-testimages/quay.io/prometheus-operator/prometheus-operator:v0.46.0
* gcr.io/k8s-testimages/quay.io/prometheus-operator/prometheus-config-reloader:v0.46.0

Please retest and verify that all images are in place.

Comment (s-urbaniak) [reply clusterloader2/pkg/prometheus/manifests/0prometheus-operator-deployment.yaml:34]: ok, it seems these are up-to-date now :)

**Comment by s-urbaniak** [comment]
@mborsz @wojtek-t can you PTAL?

**Pull request merged**